agent/agentdata/store.go

```diff
@@ -0,0 +1,26 @@
+package agentdata
+
+import (
+	"encoding/json"
+	"os"
+)
+
+func (ad AgentData) WriteToFile(path string) error {
+	b, err := json.Marshal(ad)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(path, b, 0600)
+}
+
+func ReadFromFile(path string) (AgentData, error) {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		return AgentData{}, err
+	}
+
+	var ad AgentData
+	json.Unmarshal(b, &ad)
+	return ad, nil
```

Comment:
@volmedo Is returning the value here a problem? Am I making a whole extra copy for no reason? If I return a pointer, it seems like it makes the API weird.

Reply:
yeah, you are making a whole extra copy, but there is a good reason, and that is to not make the API weird 😄.

The flaw in the logic of my comment on `MarshalJSON` also applies here. Even it is a copy, it's just the pointer, and not the actual memory, what we are copying.